tests/test_requests.py

```diff
@@ -32,10 +33,29 @@
 from .compat import StringIO, u
 from .utils import override_environ
 
+from requests.packages.urllib3.exceptions import SNIMissingWarning
+
+# urllib3 sets this to only go off once, but we need it to
+# always fire for test_https_warnings to work
+warnings.simplefilter('always', SNIMissingWarning)
```

Comment:
Could you explain your concern a bit more?
py.test hides the spam by default, so it will only appear for test that fail, and only for https tests, which there are not many of at the moment any way.